### ISSUE TITLE
Crashplan Pro - Waiting for Connection

### ISSUE THREAD

**Issue by jasongrant15** [jlesage/docker-crashplan-pro#17]
I am having an issue running the latest release (2.0.1). I am able to sign into my account and I have files that Crashplan sees that I need to backup. It sits at "Crashplan Pro Online" Waiting for Connection. It never makes it further than this. I have crashplan running with the following line.

docker run -d \
    --name=crashplan-pro \
    -p 5800:5800 \
    -p 5900:5900 \
    -v /docker/appdata/crashplan-pro:/config:rw \
    -v /share:/storage:ro \
    jlesage/crashplan-pro

**Comment by jlesage** [comment]
Did you try to restart the container?
On your account (https://www.crashplanpro.com/console), do you see your device as online?

**Comment by StephanRoemer** [comment]
I indeed have the same issue: Unable to sign in. Unknown error.

**Comment by jasongrant15** [comment]
StephanRoemer, That is a different issue, if you simply hit continue, it should log you in.

 I have restarted the container, device shows as offline. I can get it to go online for a second and then it just goes to offline. It's almost as though my container can not see the crashplan site. I have tested telnet to crashplan without issue.

**Comment by jlesage** [comment]
Before using version 2.0.1, what were you using?

**Comment by jasongrant15** [comment]
I was using the final release of the 1.X branch.

**Comment by StephanRoemer** [comment]
Oh you are right Jason! I had to delete docker settings of version 1.x, though. 
It works now, thanks!

**Comment by jlesage** [comment]
@jasongrant15, can you have a look at log/service.log.0 to see if there is any error that could point to the problem?

**Comment by gunner007cd** [comment]
The issue I've been getting trying to adopt, is "There was a problem connecting to the server, please try again."

**Comment by jasongrant15** [comment]
New issues should probably go under a new ticket to keep things clean.

**Comment by StephanRoemer** [comment]
I somehow think this is all related. I could login, but now, I have a similar connection issue as well.

**Comment by jasongrant15** [comment]
This is what I see at the moment, this is after de-authorizing and attempting to sign back in. 


[12.12.17 20:19:36.769 INFO  859848511-79 backup42.service.ui.UIController] UserActionRequest: LoginRequestMessage[826046994331130951] LOGIN: username=user@orgunit.com, password=****, serverAddress=central.crashplanpro.com:443
[12.12.17 20:19:36.769 INFO  859848511-79 backup42.service.peer.Authorizer] AUTH:: LOGIN, username=XXXXXX, passwordExists=true, requestMessage=LoginRequestMessage[826046994331130951] LOGIN: username=user@orgunit.com, password=****, serverAddress=central.crashplanpro.com:443, register=false, firstName=null, lastName=null, registrationKey=null, waitTimeMillis=120000
[12.12.17 20:19:36.769 INFO  41837_AUTH-3 backup42.service.peer.Authorizer] AUTH:: *** START *** with Authorizer@868665189[ LOGIN, username=user@orgunit.com, passwordExists=true]
[12.12.17 20:19:36.769 INFO  41837_AUTH-3 up42.service.peer.PeerController] Attempting to connect to CPC. delay=120000
[12.12.17 20:20:35.362 WARN  61348_AUTH-2 up42.service.peer.PeerController]   Couldn't connect to CPC using authority fallbacks, nothing else to do.
[12.12.17 20:20:35.363 INFO  61348_AUTH-2 backup42.service.peer.Authorizer] AUTH:: Authority address: central.crashplan.com/216.17.8.55
[12.12.17 20:20:35.363 INFO  61348_AUTH-2 backup42.service.peer.Authorizer] AUTH:: Unable to authorize, cannot connect to the authority
[12.12.17 20:20:35.365 INFO  61348_AUTH-2 backup42.service.peer.Authorizer] AUTH:: Error = CPErrors.Global.CPC_UNAVAILABLE : [central.crashplan.com:443]
[12.12.17 20:20:35.365 INFO  61348_AUTH-2 backup42.service.peer.Authorizer] AUTH:: Send login response to the UI. LoginResponseMessage@1427239921[ session=826046689740774471, errors=[CPErrors.Global.CPC_UNAVAILABLE : [central.crashplan.com:443]] ]{}
[12.12.17 20:20:35.365 INFO  61348_AUTH-2 backup42.service.ui.UIController] UserActionResponse: LoginResponseMessage@1427239921[ session=826046689740774471, errors=[CPErrors.Global.CPC_UNAVAILABLE : [central.crashplan.com:443]] ]{}
[12.12.17 20:20:35.365 INFO  61348_AUTH-2 backup42.service.peer.Authorizer] AUTH:: *** END *** Failed after 4min 0sec

**Comment by jasongrant15** [comment]
here is more detail, attached as file.
[crashplan.txt](https://github.com/jlesage/docker-crashplan-pro/files/1553191/crashplan.txt)

**Comment by jasongrant15** [comment]
[service.log.0.txt](https://github.com/jlesage/docker-crashplan-pro/files/1553211/service.log.0.txt)
Full File after login, unable to add crashplan pro online as a backup location.

**Comment by StephanRoemer** [comment]
Not sure if this will help you, but in my DiskStation I changed the network settings to "use same network as Docker Host" and now everything works.

**Comment by jasongrant15** [comment]
I figured out the issue.

I am running a QNAP device and here is what it turned out to be.

I opened the conf/my.service.xml file. Inside there is a line 

<location>0.0.0.0:4242</location> This should have been set to 127.0.0.1:4242. Modifying this, stopping and starting the container fixed my connectivity issues.

**Comment by jlesage** [comment]
Cool I will be able do to a fix for this.  Good catch!

**Comment by jlesage** [comment]
@jasongrant15,  can you give the output of this command:
`docker exec <container name> netstat -l`

**Comment by jasongrant15** [comment]
[~] # docker exec crashplan-pro-cp1 netstat -l
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State
tcp        0      0 0.0.0.0:5800            0.0.0.0:*               LISTEN
tcp        0      0 0.0.0.0:5900            0.0.0.0:*               LISTEN
tcp        0      0 :::5800                 :::*                    LISTEN
tcp        0      0 :::5900                 :::*                    LISTEN
tcp        0      0 localhost:4244          :::*                    LISTEN
Active UNIX domain sockets (only servers)
Proto RefCnt Flags       Type       State         I-Node Path
unix  2      [ ACC ]     STREAM     LISTENING     8819481 @/tmp/.X11-unix/X0
unix  2      [ ACC ]     STREAM     LISTENING     8850704 /tmp/crashplan-app.sock
unix  2      [ ACC ]     STREAM     LISTENING     8819482 /tmp/.X11-unix/X0
unix  2      [ ACC ]     STREAM     LISTENING     8820923 /var/run/s6/fdholderd-socket

**Comment by jlesage** [comment]
On 2 of my setup, I have `<location>0.0.0.0:4242</location>` and no issue.
I'm wondering if changing the value have any effect, since port 4242 is not used...  Can you try to revert back to the default value to see you connectivity issue comes back?

**Comment by jlesage** [comment]
Connectivity or login related issues seems to disappear after multiple attempts.  So maybe your issue got fixed by itself not because you changed a value in the file, but because you restarted the container.

**Comment by jasongrant15** [comment]
Possible, all is good now, I'll keep you up to date. For now, lets consider this resolved.

**Comment by maluht** [comment]
@jlesage, first of all thanks for this great container. 

I have a weird problem related to connection as well. I keep getting disconnected from the Online server every 2-3 min.

/22/18 11:16 PM	[PRO Online Backup Set] Stopped backup to CrashPlan PRO Online in 0h:00m:47s: 0 files (982MB) backed up, 0MB encrypted and sent (Effective rate: 177.1Mbps)
03/22/18 11:16 PM	- Reason for stopping backup: The backup destination was disconnected
03/22/18 11:16 PM	- Unable to backup 1 file (next attempt within 30 minutes)
03/22/18 11:17 PM	[PRO Online Backup Set] Starting backup to CrashPlan PRO Online: 81,299 files (869.10GB) to back up

Any help would be great!

**Comment by jlesage** [comment]
For this kind of issue, I would recommend to contact CrashPlan's support.  This doesn't seem do be a problem related to the container.
You can have a quick chat with them.  Let me know if anything useful is found!

**Comment by maluht** [comment]
I cleared the cache folder and that fixed the problem. But thanks for your input.

**Comment by jlesage** [comment]
Thanks for the feedback.  I will add this case to the `Troubleshooting` section.

**Comment by maluht** [comment]
No trouble and again great work with the container :)

**Comment by maluht** [comment]
@jlesage, sadly it seems the fix only worked for one day, now its back to the same problem. Client installed on the machine works fine. Any ideas?

**Comment by bigtfromaz** [comment]
It looks like you might be having a network problem.  Have you opened a command prompt in the container and verified network settings and access to the internet?  Does nslookup properly resolve host names to addresses?  Do you have a firewall that may be getting in the way?


From my phone, so this message may be more terse than normal

________________________________

**Comment by maluht** [comment]
Yes, I have a ping running to google that never fails. I might have narrowed it down to a specific file that is creating the issue. I have now removed all files and taking my 1tb data folder by folder. This also explains why it helped clearning the cache, since it then started in a new random order when uploading the data. Ill keep you informed when I figure out more.
